syntax = "proto3";

message ServerConfig {
  // The collection of External WioLinks
  map<string, WioLinkConfig> wioLinks = 1;
  repeated ArduinoConfig arduinos = 2;
  PortConfig ports = 3;
}

message WioLinkConfig {
	string rootUrl = 1;
	string accessToken = 2;
	map<string, GroveSensor> sensors = 3;
}

message GroveSensor {
	string apiMethod = 1;
}

message ArduinoConfig {
	string comPort = 1;
	int32 baudRate = 2;
}

message PortConfig {
  // Sensor Servers
  int32 frontEndPort = 1;
  int32 arduinoPort = 2;
  int32 wioPort = 3;

  // Event Servers
  int32 pushFrontEndPort = 4;
  int32 pushPort = 5;

  // Database
  int32 daoPort = 6;
}
